app/jobs/scheduled/clean_up_unused_staged_users.rb

```diff
@@ -6,11 +6,21 @@ class CleanUpUnusedStagedUsers < Jobs::Scheduled
     def execute(args)
       destroyer = UserDestroyer.new(Discourse.system_user)
 
-      User.joins(:user_stat)
-          .where(staged: true)
-          .where("users.created_at < ?", 1.year.ago)
-          .where("user_stats.post_count = 0")
```

Comment:
I think we need to add a ensure consistency job for `user_stats.post_count` because there are users that have a `user_stats.post_count` of `0` but `user.posts.count` shows otherwise.